website/content/docs/ecs/terraform/secure-configuration.mdx

```diff
@@ -84,25 +130,35 @@ resource "aws_secretsmanager_secret_version" "gossip_key" {
+  consul_server_ca_cert_arn = aws_secretsmanager_secret.ca_cert.arn
+  gossip_key_secret_arn     = aws_secretsmanager_secret.gossip_key.arn
 
-  acls                           = true
-  consul_client_token_secret_arn = module.acl_controller.client_token_secret_arn
-  acl_secret_name_prefix         = "consul-ecs"
+  acls                      = true
+  consul_http_addr          = "https://consul-server.example.com:8501"
+  consul_https_ca_cert_arn  = aws_secretsmanager_secret.ca_cert.arn
 }
 ```
 
+The following table explains the `mesh-task` input variables relevant to a secure configuration:
+
+| Input Variable              | Type     | Description                                                                                                                                                                                                                                |
+| -----------------------     | -------- | -------------------------------------------------------------------------------------------------------------------                                                                                                                        |
+| `tls`                       | boolean  | If true, TLS is enabled for RPC communication with the Consul servers.                                                                                                                                                                     |
+| `consul_server_ca_cert_arn` | string   | The Secrets Manager secret containing the CA certificate for RPC communication with the Consul servers when TLS is enabled.                                                                                                                |
+| `gossip_key_secret_arn`     | string   | The Secrets Manager secret containing Consul's gossip encryption key.                                                                                                                                                                      |
+| `acls`                      | boolean  | If true, ACLs are enabled.                                                                                                                                                                                                                 |
+| `consul_http_addr`          | string   | The Consul server address. Required when `acls = true` in order to log in to Consul's AWS IAM auth method to obtain ACL tokens.                                                                                                            |
+| `consul_https_ca_cert_arn`  | string   | (optional) The Secrets Manager secret containing the CA cert for HTTPS communication with Consul servers. Required if the server's certificate is self-signed or signed by an internal CA. This is not required for Consul servers in HCP. |
+
 Now you can deploy your services! Follow the rest of the steps in the [Installation instructions](/docs/ecs/terraform/install#task-module)
```

Comment:
Should we have a section on how to control which tasks can be part of the mesh? Maybe talk about the security posture? So for example: "anyone with the ability to create iam roles and ecs tasks can create a task that joins the mesh wth any name" (not sure if that's true, but you get the idea. Also this could be done in a subsequent PR if you think it's a good idea)

Reply:
Looks great!